src/worker.ts

```diff
@@ -32,6 +33,22 @@ export class RouteInfo {
 
     return this.inner.nameForRoute(linestring);
   }
+
+  speedLimitForRoute(waypoints: Waypoint[]): string {
+    if (!this.inner) {
+      throw new Error("Need to loadFile before using RouteInfo");
```

Comment:
Done. I'd like to maybe surface a second progress bar (but where?) and disable the controls that try to invoke route info until it's ready. Not sure how yet, so a better error message is a good interim

Reply:
UX-wise that's a little confusing to me, since the snap tool usually finishes loading first (smaller file) and the tool is usable before the route info is ready. I was thinking about putting the progress bar near the two dropdowns that let you enable speed limit, but maybe that's also a bit confusing.

I also want to overhaul how we do progress bars; the one for the route snapper is scary old non-TS code. I started https://github.com/acteng/atip/tree/progress, but could use help polishing / figuring out all the edge cases. Oddly enough, I've had trouble finding a simple library on NPM to do this kind of thing!

Reply:
#181